x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/crawler/crawl_custom_settings_flyout/crawl_custom_settings_flyout.tsx

```diff
@@ -62,22 +71,61 @@ export const CrawlCustomSettingsFlyout: React.FC = () => {
             {i18n.translate(
               'xpack.enterpriseSearch.crawler.crawlCustomSettingsFlyout.flyoutHeaderDescription',
               {
-                defaultMessage: 'Set up a one-time crawl with custom settings.',
+                defaultMessage: 'Set up a one-time crawl or multiple crawling custom settings.',
               }
             )}
           </p>
         </EuiText>
       </EuiFlyoutHeader>
+
       <EuiFlyoutBody>
         {isDataLoading ? (
           <Loading />
         ) : (
           <>
-            <CrawlCustomSettingsFlyoutCrawlDepthPanel />
+            <CrawlCustomSettingsFlyoutCrawlTypeSelection />
             <EuiSpacer />
-            <CrawlCustomSettingsFlyoutDomainsPanel />
-            <EuiSpacer />
-            <CrawlCustomSettingsFlyoutSeedUrlsPanel />
+            {crawlType === CustomCrawlType.ONE_TIME ? (
+              <>
+                {/* <CrawlCustomSettingsFlyoutCrawlDepthPanel />
+                <EuiSpacer />
+                <CrawlCustomSettingsFlyoutDomainsPanel />
+                <EuiSpacer />
+                <CrawlCustomSettingsFlyoutSeedUrlsPanel /> */}
+              </>
+            ) : (
+              <>
+                <CrawlCustomSettingsFlyoutMultipleCrawlTabs />
+                {crawlerConfigs.map((config, index) => {
+                  if (index === activeCrawlerConfigTab) {
```

Comment:
I would suggest replacing `index` with a new field, i.e. `selectedConfig` to remove the need for passing it to every action, so your action is responsible to find and change the correct configuration. 

If there is a guaranteed unique field in config, you can use it to identify and apply changes.